File: docs/code_examples/point3d_random.rs
//! Log some random points with color and radii.

use rand::{distributions::Uniform, Rng as _};
use rerun::{archetypes::Points3D, components::Color, RecordingStreamBuilder};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let (rec, storage) = RecordingStreamBuilder::new("rerun_example_points3d_random").memory()?;

    let mut rng = rand::thread_rng();
    let dist = Uniform::new(-5., 5.);

    rec.log(
        "random",
        &Points3D::new((0..10).map(|_| (rng.sample(dist), rng.sample(dist), rng.sample(dist))))
            .with_colors((0..10).map(|_| Color::from_rgb(rng.gen(), rng.gen(), rng.gen())))
            .with_radii((0..10).map(|_| rng.gen::<f32>())),
    )?;

    rerun::native_viewer::show(storage.take())?;
    Ok(())
}

File: docs/code-examples/all/point3d_random.py
"""Log some random points with color and radii."""
import rerun as rr
from numpy.random import default_rng

rr.init("rerun_example_points3d_random", spawn=True)
rng = default_rng(12345)

positions = rng.uniform(-5, 5, size=[10, 3])
colors = rng.uniform(0, 255, size=[10, 3])
radii = rng.uniform(0, 1, size=[10])

rr.log("random", rr.Points3D(positions, colors=colors, radii=radii))
